internal/graph/check_fast_path.go

```diff
@@ -24,80 +24,13 @@ const IteratorMinBatchThreshold = 1000
 const BaseIndex = 0
 const DifferenceIndex = 1
 
-type fastPathSetHandler func(context.Context, *iteratorStreams, chan<- *iteratorMsg)
-
-type iteratorMsg struct {
-	iter storage.TupleKeyIterator
-	err  error
-}
-
-type iteratorStream struct {
-	idx    int
-	buffer storage.TupleKeyIterator
-	done   bool
-	source chan *iteratorMsg
-}
-
-type iteratorStreams struct {
-	streams []*iteratorStream
-}
-
-// getActiveStreamsCount will return the active streams from the last time getActiveStreams was called.
-func (s *iteratorStreams) getActiveStreamsCount() int {
-	return len(s.streams)
-}
-
-// Stop will drain all streams completely to avoid leaving dangling resources
-// NOTE: caller should consider running this in a goroutine to not block.
-func (s *iteratorStreams) Stop() {
-	for _, stream := range s.streams {
-		if stream.buffer != nil {
-			stream.buffer.Stop()
-		}
-		for msg := range stream.source {
-			if msg.iter != nil {
-				msg.iter.Stop()
-			}
-		}
-	}
-}
-
-// getActiveStreams will return a list of the remaining active streams.
-// To be considered active your source channel must still be open.
-func (s *iteratorStreams) getActiveStreams(ctx context.Context) ([]*iteratorStream, error) {
-	for _, stream := range s.streams {
-		if stream.buffer != nil || stream.done {
-			// no need to poll further
-			continue
-		}
-
-		select {
-		case <-ctx.Done():
-			return nil, ctx.Err()
-		case i, ok := <-stream.source:
-			if !ok {
-				stream.done = true
-				break
-			}
-			if i.err != nil {
-				return nil, i.err
-			}
-			stream.buffer = i.iter
-		}
-	}
-	// TODO: in go1.23 compare performance vs slices.Collect
-	// clean up all empty entries that are both done and drained
-	s.streams = slices.DeleteFunc(s.streams, func(entry *iteratorStream) bool {
-		return entry.done && entry.buffer == nil
-	})
-	return s.streams, nil
-}
+type fastPathSetHandler func(context.Context, *iterator.Streams, chan<- *iterator.Msg)
 
 // fastPathDirect assumes that req.Object + req.Relation is a directly assignable relation, e.g. define viewer: [user, user:*].
 // It returns a channel with one element, and then closes the channel.
 // The element is an iterator over all objects that are directly related to the user or the wildcard (if applicable).
 func fastPathDirect(ctx context.Context,
```

Comment:
Nit: could you put the signature of `fastPathDirect` in one line? Makes it more readable.